x-pack/plugins/alerting/server/task_runner/create_execution_handler.test.ts

```diff
@@ -525,9 +515,88 @@ describe('Create Execution Handler', () => {
       alertExecutionStore,
     });
 
-    expect(alertExecutionStore.numberOfTriggeredActions).toBe(2);
-    expect(alertExecutionStore.numberOfScheduledActions).toBe(3);
-    expect(alertExecutionStore.triggeredActionsStatus).toBe(ActionsCompletion.PARTIAL);
+    expect(alertExecutionStore.getNumberOfTriggeredActions()).toBe(2);
+    expect(alertExecutionStore.getTriggeredActionsStatus()).toBe(ActionsCompletion.PARTIAL);
+    expect(createExecutionHandlerParams.logger.debug).toHaveBeenCalledTimes(1);
     expect(actionsClient.enqueueExecution).toHaveBeenCalledTimes(2);
   });
+
+  test('Skips triggering actions for a specific action type when it reaches the limit for that specific action type', async () => {
+    const executionHandler = createExecutionHandler({
+      ...createExecutionHandlerParams,
+      actionsConfigMap: {
+        default: {
+          max: 4,
+        },
+        'test-action-type-id': {
+          max: 1,
+        },
+      },
+      actions: [
+        ...createExecutionHandlerParams.actions,
```

Comment:
nit: I don't think we need to add the actions from here - it makes me search to find what they are :-)